.github/workflows/build-test-deploy.yml

```diff
@@ -491,14 +557,34 @@ jobs:
           name: ${{  env.NIX_CACHIX }}
 
       - run: |
-          # start network and wait to start writing log and write success start into log
-          ( nix run .#devnet-dali 2>&1 & ) | tee logfile.log &
-          until test -f logfile.log; do sleep 1 && echo "waiting network start "; done;
-          ( tail --follow --lines=0  logfile.log & ) | grep --max-count=1 "POLKADOT LAUNCH COMPLETE"
+          # ideally it does:
+          # start first process
+          # start second if and only if first stdout specific string
+          # first stdout await has timeout, any stderr or exit end stdout timeout and shell exits non zero
+          # if first or second second stderr or non zero exited, then end second and first end and shell non zero exit          
+          # if second timeouted, then in shell non zero exit
+          # more complicated, stderr by regex X times before exit
+          # if second ended success, first stopped gracefully
+          # 
+          # TODO: handle
+          # 🧹 Resolving parachain id...
+          # /nix/store/rfrhfi70ijykigwyaswrwlmfqbxw570g-polkadot-v0.9.27/bin/polkadot build-spec --chain=rococo-local --disable-default-bootnode 
+          # 2022-08-25 02:01:57 Building chain spec    
+          # ⚠ failed to parse the chain spec
+
+          ( nix run .#devnet-dali 2>&1 & ) | tee devnet-dali.log &
+          until test -f devnet-dali.log; do sleep 1 && echo "waiting network start"; done;
+          ( tail --follow --lines=0  devnet-dali.log & ) | grep --max-count=1 "POLKADOT LAUNCH COMPLETE"
           
           echo "PATH=$(pwd):$PATH" >> $GITHUB_ENV
           cd integration-tests/runtime-tests          
-          export ENDPOINT=127.0.0.1:9988 && export ENDPOINT_RELAYCHAIN=127.0.0.1:9944 && npm install -q && npm run test_short
+          npm install -q
+          ( export ENDPOINT=127.0.0.1:9988 && export ENDPOINT_RELAYCHAIN=127.0.0.1:9944 && npm run test_short 2>&1 &) | tee runtime-tests.log & RUNTIME_TESTS_PID=$!
+          until test -f runtime-tests.log; do sleep 1 && echo "waiting tests start"; done;
+          ( tail --follow --lines=0 runtime-tests.log & ) | ( grep --max-count=1 "API-WS: disconnected from" > stop.log & )
+          (while : ; do if test $(wc --lines stop.log | cut --delimiter " " --fields 1) -gt 0; then kill -9 $RUNTIME_TESTS_PID && echo "Failed" && exit 42; fi; sleep 1; done ) & 
```

Comment:
```suggestion
          (while : ; do if test $(stat --format "%s" stop.log) -gt 0; then kill -9 $RUNTIME_TESTS_PID && echo "Failed" && exit 42; fi; sleep 1; done ) & 
```

Reply:
nice, thank you. will do in next PR.

Reply:
i cannot merge now as afraid to fuck up with merged main. need to get how to rebases from merged commits